src/text_atlas.rs

```diff
@@ -68,20 +76,121 @@ impl InnerAtlas {
 
         loop {
             let allocation = self.packer.allocate(size);
+
             if allocation.is_some() {
                 return allocation;
             }
 
             // Try to free least recently used allocation
-            let (_, value) = self.glyph_cache.pop_lru()?;
-            self.packer
-                .deallocate(value.atlas_id.expect("cache corrupt"));
+            let (mut key, mut value) = self.glyph_cache.peek_lru()?;
+
+            while value.atlas_id.is_none() {
+                let _ = self.glyph_cache.pop_lru();
+
+                (key, value) = self.glyph_cache.peek_lru()?;
+            }
+
+            if self.glyphs_in_use.contains(&key) {
```

Comment:
Do we need to check `self.glyphs_in_use` for each of the keys in the loop too?

Reply:
Yes, you are right. Given we can `grow` now, we should never `pop` any glyphs that are in use. Fixed in b979122dfa80e2072f0948aa227712201e173cc4.